--- EMCpower-5.0.0/PowerPath~	2007-02-07 10:06:21.000000000 +0200
+++ EMCpower-5.0.0/PowerPath	2006-09-04 16:10:03.000000000 +0300
@@ -15,6 +15,9 @@
 
 # @(#) $Header$
 
+# Source function library.
+. /etc/rc.d/init.d/functions
+
 
 # Functions to preserve exceptional command status return values
 
@@ -39,10 +42,10 @@
         then
         case "$rc_script_stat" in
             0)
-              echo "`eval_gettext \" done\"`"
+			  nls " done"
               ;;
             *)
-              echo "`eval_gettext \" failed\"`"
+			  nls " failed"
               ;;
         esac
     fi
@@ -106,19 +109,19 @@
                         err_stop=`/sbin/vgchange -a n $i  2>&1 | grep "open logical volume"`
                         if [ "$err_stop" != "" ]; then
                                 if [ "$open_vol" = "" ]; then
-                                        echo "`eval_gettext \"Following LVM volume groups are in use:\"`"
+										nls "Following LVM volume groups are in use:"
                                         open_vol=$err_stop
                                 fi
 			var1=$i
 			var2=`echo $pv`
-                        echo "`eval_gettext \"      Volume Group: \\$var1 (\\$var2)\"`"
+                        echo "`nls "      Volume Group: \\$var1 (\\$var2)"`"
                         fi
                 fi
         done
     fi
         if [ "$open_vol" != "" ]; then
-            echo "`eval_gettext \"These open logical volume devices use LUNs from Powerpath managed devices,\"`"
-            echo "`eval_gettext \"Please re-issue the command after closing these volumes.\"`"
+			nls "These open logical volume devices use LUNs from Powerpath managed devices,"
+			nls "Please re-issue the command after closing these volumes."
 	return 1
         fi
  return 0
@@ -154,9 +157,9 @@
     rc_check
 
     if rc_status ; then
-        echo -e "`eval_gettext \"\nSuccessfully started 32-bit emulation library\"`"
+	nls "\nSuccessfully started 32-bit emulation library"
     else
-        echo -e "`eval_gettext \"\nError in starting 32-bit emulation library\"`"
+	nls "\nError in starting 32-bit emulation library"
     fi
 }
 
@@ -166,8 +169,8 @@
 {
     if [ -f /etc/emcp_devicesDB.dat ]; then 
         if [ -f /etc/emcp_devicesDB.idx ]; then
-            /etc/opt/emcpower/emcpmgr map -p > /dev/null 2>&1
+            /sbin/emcpmgr map -p > /dev/null 2>&1
             rc_check
         fi
     fi
 
@@ -206,10 +209,10 @@
 
     /sbin/powermt load		> /dev/null 2>&1
     rc_check
-    /etc/opt/emcpower/emcpmgr map	> /dev/null 2>&1
+    /sbin/emcpmgr map	> /dev/null 2>&1
     rc_check
-    /etc/opt/emcpower/powercf -C	> /dev/null 2>&1
+    /sbin/powercf -C	> /dev/null 2>&1
     /sbin/powermt save		> /dev/null 2>&1
     rc_check
     /sbin/powermt register	> /dev/null 2>&1
     rc_check
@@ -224,14 +227,14 @@
 {
     ps -C naviagent >> /dev/null
     if [ $? -eq 0 ]; then
-        echo "`eval_gettext \"Navisphere agent is running.\"`"
-        echo  "`eval_gettext \"Please stop the agent and then re-issue \\$script_name stop.\"`"
+	nls "Navisphere agent is running."
+        eval echo "$(nls \"Please stop the agent and then re-issue \\$script_name stop.\")"
         return 1
     fi
 
     ps -C powermt >> /dev/null
     if [ $? -eq 0 ]; then
-        echo "`eval_gettext \"The powermt command is running.\"`"
-        echo "`eval_gettext \"Please stop powermt and then re-issue \\$script_name stop.\"`"
+	nls "The powermt command is running."
+        echo "`eval nls \"Please stop powermt and then re-issue \\$script_name stop.\"`"
         return 1
     fi
@@ -260,6 +263,6 @@
         /sbin/modprobe -q -s --first-time $d
         rc_check
         if [ $? -ne 0 ] ; then
-            echo "`eval_gettext \"PowerPath could not load module \\$d\"`"
+            echo "`eval nls \"PowerPath could not load module \\$d\"`"
             break
         fi
@@ -288,8 +291,8 @@
 
 case "$1" in
     start)
 
-        echo -n "`eval_gettext \"Starting \\$script_name: \"`"
+        eval echo -n "$(nls "Starting \$script_name: ")"
 
         ###
         # Start the 32-bit emulation for ia64
@@ -411,7 +414,7 @@
                     unload_drivers
                 fi
             else
-                echo "`eval_gettext \"PowerPath: unable to load PowerPath modules.\"`"
+				nls "PowerPath: unable to load PowerPath modules."
                 unload_drivers
             fi
         fi
@@ -419,11 +422,11 @@
         ;;
 
     stop)
-        echo -n "`eval_gettext \"Stopping \\$script_name: \"`"
+        eval echo -n "$(nls "Stopping \$script_name: ")"
         lsmod | grep -w '^emcp' > /dev/null
         if test $? -ne 0 ;
         then
-                echo "`eval_gettext \"PowerPath is not running\"`"
+				nls "PowerPath is not running"
         else
             #
             # Tresspass can happen if new devices are added or some devices 
@@ -436,9 +439,9 @@
             if ok_to_stop; then
                 /sbin/powermt save > /dev/null 2>&1
                 rc_check
-                /etc/opt/emcpower/emcpmgr unmap > /dev/null 2>&1
+                /sbin/emcpmgr unmap > /dev/null 2>&1
                 rc_check
                 /sbin/powermt remove dev=all 2> /var/tmp/.pp_exit
                 devr=$?
                 cat /var/tmp/.pp_exit | grep "not found" > /dev/null 2>&1
                 cdevs=$?
@@ -477,7 +480,7 @@
         ;;
     *)
         var=$0
- 	echo "`eval_gettext \"Usage: \\$var {start|stop}\"`"
+		eval echo "$(nls "Usage: \$var {start\|stop}")"
 	;;
 esac
 
